components/shared_code/src/shared/database/measurements.py

```diff
@@ -27,6 +27,9 @@ def latest_successful_measurement(
 
 def insert_new_measurement(database: Database, measurement: Measurement) -> Measurement:
     """Insert a new measurement."""
+    if latest := latest_measurement(database, measurement.metric):  # pragma: no feature-test-cover
```

Comment:
The name `latest` was a bit confusing, although I'm not sure that naming it `previous` would make things better :(

Reply:
`lastest_measurement_soon_to_be_the_previous_measurement` :-)